include/git2/diff.h

```diff
@@ -444,6 +444,11 @@ typedef struct {
 	 * Defaults to "b".
 	 */
 	const char *new_prefix;
+
+	/** Skip files in the diff that are excluded by the `sparse-checkout` file.
+	 * Set to 1 to skip sparse files, 0 otherwise
+	 */
+	int skip_sparse_files;
```

Comment:
This one might require some discussion on how flags are handled.
There is a flags field, but it is almost full. Only  `(1u << 27)` is available.

Most of the flags are in enums, I think the reason for this is that it provides some type safety while still being passable as arguments to functions.
In this particular case the flags are never used as a parameter, the user passes the git_diff_options struct instead.
This means that it would be possible to specify the flags in bitfields as `unsigned int skip_sparse_files:1;`
Unfortunately we can't split out the existing enum to a regular bitfield without  being incompatible but IMO that would be a good thing to do for flags passed in a larger struct like this.

Reply:
Pinging in @ethomson since he probably have given some more thought on what to do when we run out of flags.